vitest.config.ts

```diff
@@ -0,0 +1,38 @@
+import { defineConfig } from 'vitest/config';
+import react from '@vitejs/plugin-react';
+import path from 'path';
+
+export default defineConfig({
+  plugins: [react()],
+  test: {
+    setupFiles: ['./src/__tests__/setup-tests.ts'],
+    globals: true,
+    environment: 'jsdom',
+    coverage: {
+      provider: 'v8',
+      reporter: ['text', 'json', 'html'],
+      include: ['src/**/*.{js,jsx,ts,tsx}'],
+      exclude: [
+        'src/**/*.test.{js,jsx,ts,tsx}',
+        'src/**/*.spec.{js,jsx,ts,tsx}',
+        'src/index.{js,jsx,ts,tsx}',
+        'src/setupTests.{js,ts}',
```

Comment:
1. exclude main.tsx (there is no logic there)
2. you don't have `setupTests.{js,ts}` files, but you do have `setup-tests.ts`
3. exclude folder with types

```suggestion
        'src/main.tsx',
        './src/__tests__/setup-tests.ts',
        './src/sources/types',
```